infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -85,32 +130,43 @@ def backup_database_on_interval(seconds):
     )
     while True:
         time.sleep(seconds)
-        backup_database(make_backup_fpath())
+        backup_database(make_object_name())
 
 
 ###################
 # --- Restore --- #
 ###################
 
 
-def get_most_recent_backup_fpath():
+def download_backup_object(session, backup_bucket_name, backup_obj_name, backup_fpath):
+    """Downloads the object_name backup from the backup_bucket_name S3 bucket."""
+    s3_client = session.client("s3")
+    s3_client.download_file(backup_bucket_name, backup_obj_name, backup_fpath)
+
+
+def list_bucket_objects(session, backup_bucket_name):
+    """Returns a list of all objects in the backup_bucket_name S3 bucket."""
+    bucket = session.resource("s3").Bucket(backup_bucket_name)
+    return [obj.key for obj in bucket.objects.all()]
```

Comment:
Answering my question further down. I'm assuming `all()` legitimately means "all" so this has _got_ to work for large numbers of objects in the bucket.

Reply:
yes. unless their documentation liberally uses the word all incorrectly and fails to mention pagination for `all()` function but not for the other functions but that seems unlikely